sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1386,7 +1387,7 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
    * Create a function database (optional) and name pair.
    */
   protected def visitFunctionName(ctx: QualifiedNameContext): FunctionIdentifier = {
-    ctx.identifier().asScala.map(_.getText) match {
+    ctx.identifier.asScala.map(_.getText) match {
```

Comment:
For completeness, could you handle this case, too? @yeshengm 
```scala
scala> sql("drop function if exists max-a").show
org.apache.spark.sql.catalyst.parser.ParseException:
mismatched input '-' expecting {<EOF>, '.'}(line 1, pos 27)

== SQL ==
drop function if exists max-a
---------------------------^^^
```

Reply:
Nope. Since qualified names can be used in expressions, we can't differentiate cases such as `a - my-func(args)` and `a - my - func(args)`.

Reply:
If this PR cannot handle `CREATE/DROP/DROP FUNCTION`. We had better be clear about that. So, please change the PR title and description clearly. This PR is not for all `identifier with hyphen`.

Reply:
Yeah. I changed the title and added a paragraph to explain.